gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -169,12 +171,25 @@ def _merge_headers(known_args, pipeline_args, pipeline_mode):
     else:
       headers |= vcf_header_io.ReadVcfHeaders(known_args.input_pattern)
 
-    _ = (headers
-         | 'MergeHeaders' >> merge_headers.MergeHeaders(
-             known_args.split_alternate_allele_info_fields)
-         | 'WriteHeaders' >> vcf_header_io.WriteVcfHeaders(
-             known_args.representative_header_file))
-
+    merged_headers = (headers
+                      | 'MergeHeaders' >> merge_headers.MergeHeaders(
+                          known_args.split_alternate_allele_info_fields))
+    if known_args.robust_header_extraction:
+      headers_from_variants = (
+          _read_variants(p, known_args)
+          | 'FilterVariants' >> filter_variants.FilterVariants(
+              reference_names=known_args.reference_names)
+          | 'ExtractHeaderFieldsFromVariants' >>
```

Comment:
Consider renaming to InferUndefinedHeaderFields (I think you can drop 'FromVariants') and the return object would be `inferred_headers` as we should distinguish between defined and inferred.

Reply:
Done